.github/workflows/build.yml

```diff
@@ -49,16 +54,134 @@ jobs:
         run: make verify
       - name: Test
         run: make test
-      - name: Derive pack version from branch name
-        run: |
-          [[ $GITHUB_REF =~ ^refs\/heads\/release/(.*)$ ]] && version=${BASH_REMATCH[1]} || version=0.0.0
-          echo "::set-env name=PACK_VERSION::$version"
-        shell: bash
       - name: Build
         run: make build
         env:
           PACK_BUILD: ${{ github.run_number }}
       - uses: actions/upload-artifact@v1
         with:
           name: pack-${{ matrix.os }}
-          path: out/${{ env.PACK_BIN }}
\ No newline at end of file
+          path: out/${{ env.PACK_BIN }}
+  release:
+    if: ${{ startsWith(github.ref, 'refs/heads/release/') }}
+    needs: test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Derive pack version from branch name
+        run: |
+          [[ $GITHUB_REF =~ ^refs\/heads\/release/(.*)$ ]] && version=${BASH_REMATCH[1]}
+          echo "::set-env name=PACK_VERSION::$version"
+        shell: bash
+      - name: Download artifacts - darwin
+        uses: actions/download-artifact@v1
+        with:
+          name: pack-darwin
+      - name: Download artifacts - linux
+        uses: actions/download-artifact@v1
+        with:
+          name: pack-linux
+      - name: Download artifacts - windows
+        uses: actions/download-artifact@v1
+        with:
+          name: pack-windows
+      - name: Package artifacts - darwin
+        run: |
+          chmod +x pack-darwin/pack
+          tar -C pack-darwin -vzcf pack-darwin.tgz pack
+      - name: Package artifacts - linux
+        run: |
+          chmod +x pack-linux/pack
+          tar -C pack-linux -vzcf pack-linux.tgz pack
+      - name: Package artifacts - windows
+        run: zip -j pack-windows.zip pack-windows/pack
+      - name: Extract lifecycle version
+        id: lifecycle_version
+        run: |
+          LIFECYCLE_VERSION=$(./pack-linux/pack report | grep 'Default Lifecycle Version:' | grep -o '[^ ]*$')
+          echo "::set-output name=version::$LIFECYCLE_VERSION"
+      - name: Extract pack help
+        id: pack_help
+        run: |
+          PACK_HELP=$(./pack-linux/pack --help)
+          PACK_HELP="${PACK_HELP//'%'/'%25'}"
+          PACK_HELP="${PACK_HELP//$'\n'/'%0A'}"
+          PACK_HELP="${PACK_HELP//$'\r'/'%0D'}"
+          echo "::set-output name=help::$PACK_HELP"
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@latest
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: v${{ env.PACK_VERSION }}
+          release_name: pack v${{ env.PACK_VERSION }}
+          draft: true
+          prerelease: false
+          body: |
```

Comment:
This is so exciting! I love that we are templating this... then it will be consistent :)